src/status_im/ui/screens/profile/user/views.cljs

```diff
@@ -25,84 +29,109 @@
+            {:keys [names preferred-name]} [:ens.main/screen]]
+    (let [username (stateofus/username preferred-name)
+          qr-width (- window-width 128)
+          name     (or username preferred-name)
+          link     (universal-links/generate-link :user :external address)]
+      [react/view
+       [react/view {:style {:padding-top 16 :padding-left 16 :padding-right 16}}
+        [qr-code-viewer/qr-code-view qr-width address]
+        (when (seq names)
+          [react/view
+           [copyable-text/copyable-text-view
+            {:label       :t/ens-usernames
+             :copied-text preferred-name}
+            [react/nested-text
+             {:style               {:line-height 22 :font-size 15
+                                    :font-family "monospace"}
+              :accessibility-label :ens-username}
+             name
+             (when username [{:style {:color colors/gray}} (str "." stateofus/domain)])]]
+           [react/view {:height 1 :margin-top 12 :margin-horizontal -16
+                        :background-color colors/gray-lighter}]])
+        [copyable-text/copyable-text-view
+         {:label       :t/chat-key
+          :copied-text address}
+         [react/text {:number-of-lines     1
+                      :ellipsize-mode      :middle
+                      :accessibility-label :chat-key
+                      :style               {:line-height 22 :font-size 15
+                                            :font-family "monospace"}}
+          address]]
+        [react/view styles/share-link-button
+         [button/button-with-icon
+          {:on-press            #(list-selection/open-share {:message link})
+           :label               (i18n/label :t/share-link)
+           :icon                :main-icons/link
+           :style               {:height 44 :margin-horizontal 0}
+           :accessibility-label :share-my-contact-code-button}]]]])))
 
-(defn- profile-icon-options-ext []
-  (conj profile-icon-options {:label  (i18n/label :t/image-remove-current)
-                              :action #(re-frame/dispatch [:my-profile/remove-current-photo])}))
+(defview my-profile-toolbar-content [{:keys [photo-path] :as contact} show?]
+  (letsubs [anim-opacity (animation/create-value 0)
+            to-hide      (reagent/atom false)]
+    {:component-did-update
+     (fn []
+       (if (and (not @to-hide) @show?)
+         (animation/start
+          (animation/timing
+           anim-opacity
+           {:toValue 1
+            :duration 200
+            :easing (.-ease (animation/easing))
+            :useNativeDriver true})
+          #(reset! to-hide true))
+         (animation/start
+          (animation/timing
+           anim-opacity
+           {:toValue 0
+            :duration 200
+            :easing (.-ease (animation/easing))
+            :useNativeDriver true})
+          #(reset! to-hide false))))}
+    ;; TODO - change this to better
```

Comment:
do you still have plans to change this?

Reply:
I was planning on it.. but haven't been able to think up something different, so ok if I leave the implementation as it is and remove the comment?

Reply:
i'm not sure why " we HAVE to deref `show` in render method for this to work" , because i only see animated view and anim-opacity, i think to make `component-did-update` working you need to pass show? derefered value not atom here `my-profile-toolbar-content [{:keys [photo-path] :as contact} show?]`